src/graph.cpp

```diff
@@ -160,26 +155,42 @@ void Graph::kruskal(){
  * @param candidate - Vertex that is going to be checked
  * @return true if the candidate vertex is adjacent to the tour's last vertex, false otherwise
  */
-bool Graph::isConnectable(Vertex *candidate) const {
-    for (Edge *e : (*tour.second.rend())->getAdj()){
+bool Graph::isConnectable(std::shared_ptr<Vertex> &candidate) const {
+    for (std::shared_ptr<Edge> &e : (*tour.course.rbegin())->getAdj()){
         if (e->getDest() == candidate) return true;
     }
     return false;
 }
 
+ /**
+  * Adds a vertex to the tour structure and updates the total distance
+  * Time Complexity: O(|E|) (average case) | O(|V|*|E|) (worst case)
+  * @param stop - Vertex to add
+  */
+void Graph::addToTour(std::shared_ptr<Vertex> &stop) {
+    if (!tour.course.empty()){
+        tour.distance += findEdge((*(tour.course).rbegin())->getId(), stop->getId())->getLength();
+    }
+    tour.course.push_back(stop);
+}
+
+/**
+ * Adds the distance between the first and last nodes so the tour "closes"
+ * Time Complexity: O(|E|) (average case) | O(|V|*|E|) (worst case)
+ */
+void Graph::tieDownTour(){
+    tour.distance += findEdge((*tour.course.begin())->getId(), (*tour.course.rbegin())->getId())->getLength();
+}
+
 /**
  * @brief Iterates through the vertex set using DFS, respecting if an edge is selected or not
- * Registers the path taken in the vertex's path attribute
  * Time Complexity: O(|V|+|E|)
  * @param source - Vertex where the DFS starts
  */
-void Graph::preorderMSTTraversal(Vertex *source){
+void Graph::preorderMSTTraversal(std::shared_ptr<Vertex> source){
     source->setVisited(true);
-    if (tour.second.size() != 0){
-        tour.first += findEdge((*(tour.second).rend())->getId(), source->getId())->getDist();
-    }
-    tour.second.push_back(source);
-    for (Edge *e : source->getAdj()){
+    addToTour(source);
+    for (std::shared_ptr<Edge> &e : source->getAdj()){
```

Comment:
We're gonna have to change this here because not only should we consider the existing edges (ones in adj) but also the ones that do no exist

Reply:
### Taken care in call